tf-modules/distribution/main.tf

```diff
@@ -205,3 +212,44 @@ resource "aws_api_gateway_deployment" "s3_credentials" {
   rest_api_id = module.thin_egress_app.rest_api.id
   stage_name  = var.api_gateway_stage
 }
+
+# Egress Api Gateway Log Group Filter
+resource "aws_cloudwatch_log_subscription_filter" "egress_api_gateway_log_subscription_filter" {
+  count           = var.log_destination_arn != null && var.log_api_gateway_to_cloudwatch ? 1 : 0
+  name            = "EgressApiGatewayCloudWatchLogSubscriptionToSharedDestination"
```

Comment:
With the name hard-coded like this, is there going to be a problem with multiple deployments to the same account / region?